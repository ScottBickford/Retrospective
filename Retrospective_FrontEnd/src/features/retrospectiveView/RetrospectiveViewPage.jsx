import React, { Fragment, useReducer, useEffect } from 'react';
import {
  retrospectiveItemReducer,
  getRetrospective,
  addFeedback
} from './retrospectiveViewActions';
import { format } from 'date-fns';
import { RetrospectiveFeedback } from './RetrospectiveFeedback';

const initialData = {
  item: null,
  error: null
};

const RetrospectiveViewPage = ({ match, history }) => {
  const [data, dispatch] = useReducer(retrospectiveItemReducer, initialData);

  useEffect(() => {
    getRetrospective(match.params.retrospectiveId, dispatch);
  }, [match.params.retrospectiveId]);

  const handleAddFeedback = feedback => {
    return new Promise((resolve, reject) => {
      addFeedback(match.params.retrospectiveId, feedback)
        .then(response => {
          getRetrospective(match.params.retrospectiveId, dispatch);
          resolve(response);
        })
        .catch(error => {
          reject(new Error(error));
        });
    });
  };

  return (
    <Fragment>
      <h2>View Retrospective</h2>
      {data.item && (
        <Fragment>
          <div style={{ display: 'inline-block' }}>
            <div style={{ float: 'left', width: '150px' }}>
              <div>
                <u>Name:</u>
              </div>
              {data.item.name}
            </div>
            <div style={{ float: 'left', width: '300px' }}>
              <div>
                <u>Summary:</u>
              </div>
              {data.item.summary}
            </div>
            <div style={{ float: 'left', width: '150px' }}>
              <div>
                <u>Date:</u>
              </div>
              {format(new Date(data.item.date), 'd MMMM yyyy')}
            </div>
          </div>
          <div style={{paddingTop:20}}>
            <div>
              <u>Participants:</u>
            </div>
            {data.item.participants.map(p => {
              return <div key={p}>{p}</div>;
            })}
          </div>
          <RetrospectiveFeedback
            feedback={data.item.feedback}
            handleAddFeedback={handleAddFeedback}
          />
          <hr />
          <a href='/'>Go back</a>
        </Fragment>
      )}
    </Fragment>
  );
};

export default RetrospectiveViewPage;
