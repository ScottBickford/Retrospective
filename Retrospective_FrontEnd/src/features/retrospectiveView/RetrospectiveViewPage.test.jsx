import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RetrospectiveViewPage from './RetrospectiveViewPage';

const match = { params: { retrospectiveId: 1 } };
describe('RetrospectiveViewPage', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<RetrospectiveViewPage match={match} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes all props to Input', () => {
    const { getByText } = render(<RetrospectiveViewPage match={match} />);
    const buttonElement = getByText(/View Retrospective/i);
    expect(buttonElement).toBeInTheDocument();
  });
});
