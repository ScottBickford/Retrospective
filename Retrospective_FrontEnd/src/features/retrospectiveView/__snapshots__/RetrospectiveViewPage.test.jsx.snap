// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetrospectiveViewPage snapshot renders 1`] = `
<h2>
  View Retrospective
</h2>
`;
